### PR TITLE
Throwing an exception on selection of a QueryableValue

### PR DESCRIPTION
**Pull request by StanleyGoldman**
- To try to make things clearer when the user does something incorrect, we wanted to throw an exception that more clearly explained what a user is supposed to do.
- Also tweaked some exceptions to get a better base exception class

### CODE REVIEW

File: Octokit.GraphQL.Core/Core/Builders/QueryBuilder.cs [hunk 1]
```diff
@@ -13,6 +13,7 @@ namespace Octokit.GraphQL.Core.Builders
     public class QueryBuilder : ExpressionVisitor
     {
         const int MaxPageSize = 100;
+        public const string CannotSelectIQueryableValueUseExceptionMessage = "Cannot directly select \'IQueryableValue<>\'. Use Single() or SingleOrDefault() to unwrap the value.";
```
Comment (grokys) [inline Octokit.GraphQL.Core/Core/Builders/QueryBuilder.cs]: Should be `CannotSelectIQueryableValueExceptionMessage`.

File: Octokit.GraphQL.Core/Core/Builders/QueryBuilder.cs [hunk 2]
```diff
@@ -248,6 +249,11 @@ protected override Expression VisitNew(NewExpression node)
 
             foreach (var arg in node.Arguments)
             {
+                if (arg.Type.IsConstructedGenericType && typeof(IQueryableValue<>) == arg.Type.GetGenericTypeDefinition())
```
Comment (grokys) [inline Octokit.GraphQL.Core/Core/Builders/QueryBuilder.cs]: Should probably do the same for `IQueryableList<>` too.

File: Octokit.GraphQL.Core/Core/Builders/QueryBuilder.cs (hunk 1 above)
Comment (StanleyGoldman) [reply Octokit.GraphQL.Core/Core/Builders/QueryBuilder.cs]: 👍

File: Octokit.GraphQL.Core/Core/Builders/QueryBuilder.cs (hunk 2 above)
Comment (StanleyGoldman) [reply Octokit.GraphQL.Core/Core/Builders/QueryBuilder.cs]: Oddly.. I'm unable to construct such a query...

**Comment by StanleyGoldman** [comment]
How do we get the same original error for `IQueryableList<>`.

**Comment by grokys** [comment]
I added a test for `IQueryableList<>` and added the code to throw when detected.

**Comment by grokys** [comment]
LGTM - there are still various scenarios this doesn't catch, such as:

```
var expression = new Query()
      .Repository("foo", "bar")
      .Issue(50);
```

But this is a good start.

**Pull request merged**